var/spack/repos/builtin/packages/py-biom-format/package.py

```diff
@@ -0,0 +1,39 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyBiomFormat(PythonPackage):
+    """The BIOM file format (canonically pronounced biome) is designed to be
+       a general-use format for representing biological sample by observation
+       contingency tables."""
+
+    homepage = "https://pypi.python.org/pypi/biom-format/2.1.6"
+    url      = "https://pypi.io/packages/source/b/biom-format/biom-format-2.1.6.tar.gz"
+
+    version('2.1.6', '1dd4925b74c56e8ee864d5e1973068de')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-h5py',       type=('build', 'run'))
```

Comment:
Missing quite a few Python dependencies:
```python
if USE_CYTHON:                                                                  
    from Cython.Build import cythonize                                          
    extensions = cythonize(extensions)  
```
It looks like `cython` can be used to optimize the build. You should add `py-cython` as a `type='build'` dependency.
```python
install_requires = ["click", "numpy >= 1.3.0", "future >= 0.16.0",              
                    "scipy >= 0.13.0", 'pandas >= 0.19.2',                      
                    "six >= 1.10.0"]                                            
                                                                                
# HACK: for backward-compatibility with QIIME 1.9.x, pyqi must be installed.    
# pyqi is not used anymore in this project.                                     
if sys.version_info[0] < 3:                                                     
    install_requires.append("pyqi")
```
Not sure the best way to handle the `pyqi` dependency, but the rest are definitely required.
```
      extras_require={'test': ["nose >= 0.10.1", "flake8"],                     
                      'hdf5': ["h5py >= 2.2.0"]                                 
                      },
```
`py-h5py` is actually the only optional dependency. It would be best to add a `+hdf5` variant that triggers this dependency.